### PR TITLE
initialize status package

### PR DESCRIPTION
**Pull request by murex971**

### CODE REVIEW

**Comment by murex971** [comment]
@supra08 would `apk-install` needed to add as a different stage.
There are some status logs on build-runner pertaining to that.
Or we'll be handling it under `build` stage only?

File: pkg/status/status.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	Failed State = iota
+	InProgress
+	Success
+)
+
+type BuildStatus struct{
+	BuildStage 	Stage
+	BuildStatus State
+	Message 	string
+}
```
Comment (supra08) [inline pkg/status/status.go]: 1. Why do we need a struct for Build? If it is just a string which is sent as a message why not just construct a string with type restraints
2. Can we make the build and configure stage message structure as similar as possible?

**Review by supra08 — CHANGES_REQUESTED** [review]

**Comment by supra08** [comment]
> @supra08 would `apk-install` needed to add as a different stage. There are some status logs on build-runner pertaining to that. Or we'll be handling it under `build` stage only?

Well yes, according to recent updates and requirements, its better to make apk-install as a different stage, after the build stage.
This would make sure that the build has passed for which gradle is responsible but the apk-install fails (if it does) for which the infra is responsible

File: pkg/status/status.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+const (
+	// Stages of build process
+	CONFIGURE   Stage = "configure"
+	APK_INSTALL Stage = "apk-install"
+	BUILD       Stage = "build"
+
+	// States of build's stages
+	FAILED     State = "FAILED"
+	INPROGRESS State = "INPROGRESS"
+	SUCCESS    State = "SUCCESS"
```
Comment (supra08) [inline pkg/status/status.go:17]: it's fine that you have used `SUCCESS` here. Just make sure to communicate to the frontend the same

**Review by supra08 — APPROVED** [review]

**Pull request merged**